js/main.js

```diff
@@ -1,20 +1,38 @@
 var $searchForm = document.querySelector('#search-form');
 var $vibesMessage = document.querySelector('.vibes-message');
 var $resultsRow = document.querySelector('.results-row');
-var $allViews = document.querySelectorAll('.view');
+var $views = document.querySelectorAll('.view');
+var $genreList = document.querySelector('#genre-list');
+var $navLogo = document.querySelector('#logo-img');
+var $appName = document.querySelector('.app-name');
 
 // view swapping function
-
-function swapViews(event) {
-  var dataViewValue = event.target.getAttribute('data-view');
-  if (dataViewValue === null) {
-    return;
-  }
-  for (var i = 0; i < $allViews.length; i++) {
-    if ($allViews[i].getAttribute('data-view') === dataViewValue) {
-      $allViews[i].className = 'view';
+function swapViews(view) {
+  for (var i = 0; i < $views.length; i++) {
+    if ($views[i].getAttribute('data-view') === view) {
+      $views[i].classList.remove('hidden');
     } else {
-      $allViews[i].className = 'view hidden';
+      $views[i].classList.add('hidden');
+    }
+  }
+}
+
+function showArtistSearch() {
+  swapViews('artist-search');
+}
+
+function showHomePage() {
+  swapViews('home-page');
+}
+
+// handle liked Artists function
+function likeArtist(event) {
+  var dataArtistIdNum = parseInt(event.target.getAttribute('data-artist-id'));
+  for (var i = 0; i < data.artists.length; i++) {
+    if (dataArtistIdNum === data.artists[i].id) {
+      $resultsRow.children[i].remove();
+      data.likedArtists.push(data.artists[i]);
+      data.artists.splice(data.artists[i], 1);
```

Comment:
Are you sure this is doing anything? It looks like you are passing an object as the first argument, which the splice method doesn't accept as an argument, it is expecting a number.